ytmusicapi/ytmusic.py

```diff
@@ -28,16 +28,17 @@ class YTMusic(BrowsingMixin, SearchMixin, WatchMixin, ExploreMixin, LibraryMixin
     """
 
     def __init__(self,
-                 auth: str = None,
+                 auth: Optional[str | dict] = None,
```

Comment:
I assume this change is only to avoid having to pass a JSON string instead of the dictionary. Not strictly needed,  just nice to have?

Reply:
Correct. It enables you to skip dumping a dict to json string only to have it immediately loaded back to a dict.

Reply:
ok, just wanted to make sure I understand the intent